presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -618,4 +625,79 @@ private static String getRenameErrorMessage(Path source, Path target)
     {
         return format("Error moving data files from %s to final location %s", source, target);
     }
+
+    public static List<String> convertPredicateToParts(TupleDomain<Column> effectivePredicate)
+    {
+        List<String> filter = new ArrayList<>();
+        effectivePredicate.getDomains().ifPresent(domains -> {
+            for (Map.Entry<Column, Domain> columnDomainEntry : domains.entrySet()) {
+                Domain domain = columnDomainEntry.getValue();
+                if (domain != null && domain.isNullableSingleValue()) {
+                    Object value = domain.getNullableSingleValue();
+                    Type type = domain.getType();
+                    filter.add(convertRawValueToString(value, type));
+                }
+                else {
+                    filter.add(PARTITION_VALUE_WILDCARD);
+                }
+            }
+        });
+        return filter;
+    }
+
+    public static String convertRawValueToString(Object value, Type type)
+    {
+        String val;
+        if (value == null) {
+            val = HIVE_DEFAULT_DYNAMIC_PARTITION;
+        }
+        else if (type instanceof CharType) {
+            Slice slice = (Slice) value;
+            val = padSpaces(slice, type).toStringUtf8();
+        }
+        else if (type instanceof VarcharType) {
+            Slice slice = (Slice) value;
+            val = slice.toStringUtf8();
+        }
+        // Types above this have only a single possible representation for each value.
+        // Types below this may have multiple representations for a single value.  For
+        // example, a boolean column may represent the false value as "0", "false" or "False".
+        // The metastore distinguishes between these representations, so we cannot prune partitions
+        // unless we know that all partition values use the canonical Java representation.
+
+        //TODO: How to incorporate this flag
+        /*
+        else if (!assumeCanonicalPartitionKeys) {
```

Comment:
This is an open question. 
What is the best way to keep supporting this flag.?
For most of our use cases partition names are strings since they are physical folders and the call never reaches here, however the spec supports using other types such as Decimals, Dates, Timestamps etc.

Reply:
I think we need to preserve this flag.

There are two ways we can proceed:

1. Keep the flag in the same place where it is right now. Prefilter the `TupleDomain<Column>` based on this flag (e.g.: if the `assumeCanonicalPartitionKeys` is `true` - keep only VARCHAR  and CHAR predicates)
2. Try to pass this flag to every single metastore implementation (this could be less practical)

Reply:
Going with Option 1 as we discussed